stat.go

```diff
@@ -864,17 +864,67 @@ type weightSorter struct {
 	w []float64
 }
 
-func (w weightSorter) Less(i, j int) bool {
-	return w.x[i] < w.x[j]
-}
-
+func (w weightSorter) Len() int           { return len(w.x) }
+func (w weightSorter) Less(i, j int) bool { return w.x[i] < w.x[j] }
 func (w weightSorter) Swap(i, j int) {
 	w.x[i], w.x[j] = w.x[j], w.x[i]
 	w.w[i], w.w[j] = w.w[j], w.w[i]
 }
 
-func (w weightSorter) Len() int {
-	return len(w.x)
+// SortWeightedLabelled rearranges the data in x along with their
+// corresponding weights and boolean labels so that the x data are sorted.
+// The data is sorted in place.
+// Weights and labels may be nil, if either is non-nil it
+// must have the same length as x.
+func SortWeightedLabelled(x []float64, labels []bool, weights []float64) {
+	if labels == nil {
```

Comment:
instead of the else, return

if labels == nil {
    SortWeighted(x, weights)
    return
}
if weights == nil {
     ...
    return 
}
rest